src/MongoDB.Driver.Core/Core/Operations/RetryableWriteOperationExecutor.cs

```diff
@@ -130,27 +128,34 @@ public static async Task<TResult> ExecuteAsync<TResult>(IRetryableWriteOperation
         }
 
         // privates static methods
+        private static bool AreRetriesAllowed<TResult>(IRetryableWriteOperation<TResult> operation, RetryableWriteContext context)
+        {
+            return IsOperationAcknowledged(operation) && DoesContextAllowRetries(context);
+        }
+
         private static bool AreRetryableWritesSupported(ConnectionDescription connectionDescription)
         {
             return
                 connectionDescription.IsMasterResult.LogicalSessionTimeout != null &&
                 connectionDescription.IsMasterResult.ServerType != ServerType.Standalone;
         }
 
-        private static bool ShouldRetryWrites<TResult>(IRetryableWriteOperation<TResult> operation, RetryableWriteContext context)
+        private static bool DoesContextAllowRetries(RetryableWriteContext context)
         {
-            var writeConcern = operation.WriteConcern;
-            if (writeConcern != null && !writeConcern.IsAcknowledged)
-            {
-                return false;
-            }
-
             return
                 context.RetryRequested &&
                 AreRetryableWritesSupported(context.Channel.ConnectionDescription) &&
                 !context.Binding.Session.IsInTransaction;
         }
 
+        private static bool IsOperationAcknowledged<TResult>(IRetryableWriteOperation<TResult> operation)
+        {
+            var writeConcern = operation.WriteConcern;
+            return 
+                writeConcern == null || // null means use server default write concern which implies acknowledged
```

Comment:
Are there tests for both sides of the OR?

Reply:
@rstam this is the only open comment thread on my end.

Reply:
I've added unit tests.